### PR TITLE
Include subpackage for SPOT data source

### PR DESCRIPTION
**Pull request by jamesvrt**
Inclusion of sub-package to perform STAC creation for SPOT imagery. Converted from https://github.com/sparkgeo/spotstac

Sub-package checklist:

- [X] Add the new subpackage as stactools_{pkg}, where {pkg} is a short name for the dataset the subpackage works with (e.g. "landsat")

- [X] Create a setup.py, requirements.txt, and README.md in the subpackage directory.
- [X] The code should exist in the stactools/{pkg}/ directory in that package subdirectory. Note that the stactools does not have an init.py (look at the other subpackages for examples).
- [X] Add the subpackage to the appropriate variables in scripts/env
- [X] Add the subpackage requirements install section to docker/Dockerfile
- [X] Add the subpackage to the appropriate tables in the README.
- [X] Add documentation for the subpackage.
- [X] Add subpackage to .readthedocs.yml install
- [X] Add the subpackage to the subpackages list in the top-level setup.py, with is_extra=True. This will allow for an install of that specific subpackage with the pip extras syntax, e.g. pip install stactools[aster].

- [X] flake8 and yapf checks run.

### CODE REVIEW

**Comment by gadomski** [comment]
Hello @JamesTownend, thanks for the PR. Two notes:
- Looks like CI is failing, try `docker/cibuild` locally to test things out if you don't want to wait for Github Actions to report.
- Did you mean to issue the PR against the `feature/pystac-v1.0` branch? No problem if so, just wanted to make sure. If you did mean to hit that branch, I force-pushed today, so you'll have to rebase onto the new branch.

**Comment by JamesTownend** [comment]
Hey @gadomski. Happy to be a contributor!

- I'll look into the CI on sparkgeo/stactools
- The PR against `feature/pystac-v1.0` was intentional as I was writing the sub-package with `pystac-v1.0` in mind. I'll look into rebasing. Should I close this PR? Sorry, still figuring out the intricacies of github collaboration.

**Comment by gadomski** [comment]
> I'll look into the CI on sparkgeo/stactools

👍🏽 

> The PR against feature/pystac-v1.0 was intentional as I was writing the sub-package with pystac-v1.0 in mind. I'll look into rebasing. Should I close this PR? Sorry, still figuring out the intricacies of github collaboration.

Nope, don't bother closing, you can rebase and force-push to this branch and things will update nicely (hopefully :-)). And no apologies necessary thanks for the contribution!

**Comment by JamesTownend** [comment]
`docker/cibuild` now builds successfully locally (I just needed to fix a typo in `setup.py`). It should be rebased too, so hopefully everything in order.

File: stactools_spot/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+fiona==1.8.13
+boto3==1.11.9
```
Comment (gadomski) [inline stactools_spot/requirements.txt:2]:
```suggestion
boto3==1.17.40
```

`stactools_threedep` [requires a newer version of boto3](https://github.com/stac-utils/stactools/blob/master/stactools_threedep/requirements.txt#L1) and pip dependency resolution was [struggling with the conflicting requirements](https://github.com/stac-utils/stactools/runs/2718764779).

File: stactools_spot/stactools/spot/stac_templates.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        "www.geobase.ca",
+    ),
+    Provider("Sparkgeo", "info@sparkegeo.com", ["processor", "host"],
+             "www.sparkgeo.com"),
+    Provider("PCI Geomatics", "info@pci.com", ["processor", "host"],
+             "www.pcigeomatics.com"),
+]
+
+SpotExtents = Extent(
+    SpatialExtent([[0.0, 0.0, 0.0, 0.0]]),
```
Comment (gadomski) [inline stactools_spot/stactools/spot/stac_templates.py:34]: What's going on here?

**Review by gadomski — CHANGES_REQUESTED** [review]
Just a couple of comments to address, thanks!

File: stactools_spot/stactools/spot/stac_templates.py (hunk 2 above)
Comment (JamesTownend) [reply stactools_spot/stactools/spot/stac_templates.py]: I've corrected the providers (Sparkgeo was providing a COG version in the repo I converted from) and included a proper spatial extent.

**Review by gadomski — APPROVED** [review]
Looks good to me, thanks for that spatial extents fix! Since this appears to be a "static" dataset, I'm assuming those extents are fixed and won't change based upon new data or anything like that?

**Comment by JamesTownend** [comment]
Yes, that's my understanding. The last acquisition was in 2010 so I don't think it makes sense to link the spatial extent to the shapefile the STAC is to be extracted from. I can include that if it better fits the `stactools` methodology.

**Comment by gadomski** [comment]
> I can include that if it better fits the stactools methodology.

No need, if it's static then hardcoding is fine IMHO.

**Review by gadomski — CHANGES_REQUESTED** [review]
@JamesTownend one thing I meant to include in my first review, but I forgot, is a request to add unit tests. Check out the existing tests directory for examples, but at minimum it'd be good to have a `CliTestCase` to check the command-line usage, and then any unit tests you feel are appropriate to test the more interesting logic in your PR. Thanks.

**Review by gadomski — APPROVED** [review]
Looks great, thank you! I've got these changes slated to go in after we merge #113, since it depends on pystac v1. We're expecting #113 to land in the next week or two. Let me know if you have deadlines that require these to be released in stactools before then, and we can try to work something out. Thank you again for the contribution and the updates!

**Comment by gadomski** [comment]
@jamesvrt FYI I'm planning to move this PR into its own repository `stactools-packages/spot` to align with the [v0.2 updates](https://github.com/stac-utils/stactools/releases/tag/v0.2.1a0). Will likely do this today or tomorrow, let me know if there's any issues with that plan.

**Comment by jamesvrt** [comment]
The timing is fine, and thanks for the rearrange

**Comment by gadomski** [comment]
Done, closing this PR. @jamesvrt if you have the time to take a look and make sure it's what you want, that'd be swell. Right now CI isn't passing because of formatting issues and because https://github.com/stac-utils/stactools/pull/113 isn't merged (PySTAC v1.0 support), so there will be some cleanup required there as well.

**Pull request closed without merging**